src/main/drivers/bus_i2c_stm32f30x.c

```diff
@@ -231,15 +231,17 @@ bool i2cWrite(uint8_t addr_, uint8_t reg, uint8_t data)
         }
     }
 
-    /* Send Register address */
-    I2C_SendData(I2Cx, (uint8_t) reg);
+    if (reg != -1) {
```

Comment:
Did you test this?
`I2C_TransferHandling` set `NBYTES` to 1, but no byte is transferred when `reg==-1`